### ISSUE TITLE
End to End Testing for Moped (Playwright)

### ISSUE THREAD

**Issue by mateoclarke** [cityofaustin/atd-data-tech#9428]
How could we replace our standard Moped manual Q&A pre-release smoke test with automated testing? Looking to dig into the libraries that do this well in a React front end stack. Maybe watch some Youtube videos and read some tutorials. Then pick a testing library and try it out.

Some options:
- [Playwright](https://github.com/microsoft/playwright)
- [Jest & Puppeteer](https://blog.logrocket.com/react-end-to-end-testing-jest-puppeteer/)
- [Cypress](https://github.com/cypress-io/cypress/tree/master/npm/react)

Goals:
- evaluate options
- pick a library to experiment with
- implement at least one E2E test from the Q&A script

**Comment by mateoclarke** [comment]
- [LinkedIn Learning: E2E testing with Jest](https://www.linkedin.com/learning/learning-end-to-end-testing-with-jest/)
- [Article: React end-to-end testing using Jest and Puppeteer](https://blog.logrocket.com/react-end-to-end-testing-jest-puppeteer/) 
- [Youtube: React Testing Crash Course](https://www.youtube.com/watch?v=OVNjsIto9xM)

Types of testing:
- Unit Testing
- Integration
- E2E

Test libraries and options in the React community:
- [Jest](https://jestjs.io/) - JS test framework and test runner (not just for React) from Facebook
- RTL ([React Test Library](https://testing-library.com/docs/react-testing-library/intro/)) - now comes default with CRA. claims to replace Enzyme (made by AirB&B). Meant for testing components but works directly with DOM nodes. Works with Jest.
- Puppeteer - Google open source project. Many of the main contributors moved to Playwright
- Cypress - only supports Firefox & Edge
- Playwright - Microsoft open source project, forked from Puppeteer. Runs across Browser Engines (Chromium, Webkit, Firefox). Made specifically for E2E testing

Comparisons:
- [Playwright vs Cypress: A Comparison
](https://www.browserstack.com/guide/playwright-vs-cypress)
- [Reddit: Cypress vs Playwright](https://www.reddit.com/r/QualityAssurance/comments/srhafv/cypress_vs_playwright/)
- https://www.reddit.com/r/reactjs/comments/spu8cm/playwright_vs_cypress/
- https://cathalmacdonnacha.com/cypress-vs-playwright-which-is-best-for-e2e-testing

**TL;DR: After more research, I'm circling back to Playwright.**
